### DISCUSSION TITLE
如何正确与tailwindcss集成

### DISCUSSION THREAD

**Discussion by seepine** [tusen-ai/naive-ui#6431] [Q&A] [answered]
例如在card使用tailwindcss时`header-class="p-2"`，样式会被card的覆盖无法生效

<img width="318" alt="image" src="https://github.com/user-attachments/assets/e12fc02e-7037-4a86-b73e-8d27d471c2c5">
<img width="432" alt="image" src="https://github.com/user-attachments/assets/e940d091-0aa9-4147-affc-6bbc9b9232db">
<img width="261" alt="image" src="https://github.com/user-attachments/assets/88ffa3c6-c3b4-48bf-83ad-90109a5c1d16">

### 按tailwindcss官方vite+vue文档引入
main.ts
```ts
import './assets/css/index.css'
```

index.css
```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Comment by seepine** [comment]
尝试了官方说的几种方式，并没有什么变化
https://www.naiveui.com/zh-CN/os-theme/docs/style-conflict

**Comment by jahnli** [comment]
`!important`

**Reply by seepine** [reply]
能否具体说明

**Reply by seepine** [reply]
p-2 是tailwindcss的类，不是自己写的，如何!important

**Reply by jahnli** [reply]
> p-2 是tailwindcss的类，不是自己写的，如何!important

`p-2!`

**Reply by seepine** [reply]
> `p-2!`

啊tailwindcss有这种用法吗？使用了并没有作用
```vue
<n-card content-class="p-2!" >
```
<img width="601" alt="image" src="https://github.com/user-attachments/assets/35135e12-f07a-4436-b33a-a9a1c91f07d2">

**Reply by seepine** [reply, marked as answer]
不过这倒是一个思路，官方文档是加在前面
<img width="711" alt="image" src="https://github.com/user-attachments/assets/5b7d2e62-7177-4b97-a1a5-e0f8e23f2e15">
<img width="585" alt="image" src="https://github.com/user-attachments/assets/e0e5a9be-9962-464f-b19b-9efb56361e98">

也可以在 `tailwind.config.ts`中添加，让所有的都加上 `!important`
https://tailwindcss.com/docs/configuration#important
```js
module.exports = {
  important: true,
}
```

**Comment by jahnli** [comment]
有，好像是前后都可以

**Reply by seepine** [reply]
难道新版本废弃了，我的版本如下，目前是只有`!p-2`生效，`p-2!`没有生效
```json
"vue": "^3.4.29",
"vite": "^5.3.1",
"tailwindcss": "^3.4.10",
"postcss": "^8.4.42",
"autoprefixer": "^10.4.20",
```

**Reply by jahnli** [reply]
> 难道新版本废弃了，我的版本如下，目前是只有`!p-2`生效，`p-2!`没有生效
> 
> ```json
> "vue": "^3.4.29",
> "vite": "^5.3.1",
> "tailwindcss": "^3.4.10",
> "postcss": "^8.4.42",
> "autoprefixer": "^10.4.20",
> ```

我最近在用 `unocss`  我不确定最新 `tailwindcss` 的改动，既然生效就前面，不过不建议所有的都加 `!import`

**Reply by seepine** [reply]
是目前就特殊地方添加!，全局怕其他地方有冲突，感谢
